Software/lib/GNSE_BSP/GNSE_bsp_clk_timer.c

```diff
@@ -59,6 +60,59 @@ int32_t GNSE_BSP_RTC_Init(void)
   return GNSE_BSP_ERROR_NONE;
 }
 
+/**
+  * @brief  Configures independent watchdog timer
+  * @note   GNSE_BSP_IWDG_Refresh should be called at a recurring point in the application
+  *         to avoid the watchdog from resetting the application.
+  * @return GNSE_BSP status
+  */
+int32_t GNSE_BSP_IWDG_Init(void)
+{
+  /* Set the timeout to its max value
+   * Reload is counted down from
+  */
+  GNSE_BSP_iwdg.Instance = IWDG;
+  GNSE_BSP_iwdg.Init.Prescaler = IWDG_PRESCALER_256;
+  GNSE_BSP_iwdg.Init.Window = IWDG_WINR_WIN;
+  GNSE_BSP_iwdg.Init.Reload = GNSE_IWDG_TIMEOUT_S;
+
+  /* Set the LSI clock */
+  /* This is also done automatically by HAL_IWDG_Init, but this function does not set the prescaler */
+  RCC_OscInitTypeDef RCC_OscInitStruct = {0};
+  RCC_OscInitStruct.OscillatorType = RCC_OSCILLATORTYPE_LSI;
+  RCC_OscInitStruct.LSIDiv = RCC_LSI_DIV128;
+  RCC_OscInitStruct.LSIState = RCC_LSI_ON;
+  HAL_Delay(IWDG_TIMER_DELAY); // Wait a bit for RCC_CSR_LSION to settle
```

Comment:
The location of calling this function seems random, can you please  explain why it's called specifically here and not in another location like before tracer init?